src/Spryker/Zed/Merchant/Persistence/MerchantRepository.php

```diff
@@ -78,4 +137,108 @@ public function hasKey(string $key): bool
+     *
+     * @return \Generated\Shared\Transfer\UrlTransfer[][]
+     */
+    public function getUrlsMapByMerchantIds(array $merchantIds): array
+    {
+        $merchantUrlTransfersMap = [];
+
+        $urlEntities = $this->getFactory()
+            ->getUrlPropelQuery()
+            ->joinWithSpyLocale()
+            ->filterByFkResourceMerchant_In($merchantIds)
+            ->find();
+
+        if (!$urlEntities->count()) {
+            return $merchantUrlTransfersMap;
+        }
+
+        foreach ($urlEntities as $urlEntity) {
+            $merchantUrlTransfersMap[$urlEntity->getFkResourceMerchant()][] = $this->getFactory()
+                ->createPropelMerchantMapper()
+                ->mapUrlEntityToUrlTransfer($urlEntity, new UrlTransfer());
+        }
+
+        return $merchantUrlTransfersMap;
+    }
+
+
+    /**
+     * @param \Orm\Zed\Merchant\Persistence\SpyMerchantQuery $merchantQuery
+     * @param \Generated\Shared\Transfer\MerchantCriteriaTransfer $merchantCriteriaTransfer
+     *
+     * @return \Orm\Zed\Merchant\Persistence\SpyMerchantQuery
+     */
+    protected function applyFilters(SpyMerchantQuery $merchantQuery, MerchantCriteriaTransfer $merchantCriteriaTransfer): SpyMerchantQuery
+    {
+        if ($merchantCriteriaTransfer->getIdMerchant() !== null) {
+            $merchantQuery->filterByIdMerchant($merchantCriteriaTransfer->getIdMerchant());
+        }
+
+        if ($merchantCriteriaTransfer->getEmail() !== null) {
+            $merchantQuery->filterByEmail($merchantCriteriaTransfer->getEmail());
+        }
+
+        if ($merchantCriteriaTransfer->getMerchantReference() !== null) {
+            $merchantQuery->filterByMerchantReference($merchantCriteriaTransfer->getMerchantReference());
+        }
+
+        if ($merchantCriteriaTransfer->getMerchantReferences()) {
+            $merchantQuery->filterByMerchantReference_In($merchantCriteriaTransfer->getMerchantReferences());
+        }
+
+        if ($merchantCriteriaTransfer->getMerchantIds()) {
+            $merchantQuery->filterByIdMerchant_In($merchantCriteriaTransfer->getMerchantIds());
+        }
+
+        if ($merchantCriteriaTransfer->getIsActive() !== null) {
+            $merchantQuery->filterByIsActive($merchantCriteriaTransfer->getIsActive());
+        }
+
+        if ($merchantCriteriaTransfer->getStatus() !== null) {
+            $merchantQuery->filterByStatus($merchantCriteriaTransfer->getStatus());
+        }
+
+        return $merchantQuery;
+    }
+
+    /**
+     * @param \Propel\Runtime\ActiveQuery\ModelCriteria $query
+     * @param \Generated\Shared\Transfer\PaginationTransfer|null $paginationTransfer
+     *
+     * @return mixed|\Propel\Runtime\ActiveRecord\ActiveRecordInterface[]|\Propel\Runtime\Collection\Collection|\Propel\Runtime\Collection\ObjectCollection
+     */
+    protected function getPaginatedCollection(ModelCriteria $query, ?PaginationTransfer $paginationTransfer = null)
+    {
+        if ($paginationTransfer !== null) {
```

Comment:
```suggestion
        if (!$paginationTransfer) { return...
```
please use early return here.